### DISCUSSION TITLE
Limited rights for "anon" account

### DISCUSSION THREAD

**Discussion by Darth-Coin** [stackernews/stacker.news#1864] [Ideas]
**Is your feature request related to a problem? Please describe.**
Is related to this comment https://stacker.news/items/575376

As far I understand the "anon" account, if you zap it with x amount of sats, all those sats are going to SN rewards pool. Correct me if I am wrong.
So is kind of meaningless to zap a large amount of sats to "anon" account. Is like rewarding yourself.
Also downzaping it is even more useless, because you achieve absolutely nothing.

If you really want to "approve" that comment, is enough 1 sat zap. I do not understand why people are zapping large amount of sats for something that achieve little.

@huumn suggested to downzap the same amount of sats. Why in the hell would suggest such thing?

**Describe the solution you'd like**
1. anon account should not be "zapeable". Is meaningless and achieve nothing, just creating a false impression that other stackers are "supporting" that account. Anybody can be anon and could post in the same thread controversial comments. is only creating more confusion.
2. anon account should pay a fix price to post: 1000 sats or more. MAKE THEM PAY if they do not want to use their SN real account and hide behind anon account. People with 2 faces are not to be trusted.
3. anon account CANNOT zap others comments/posts. I find that as a flaw in SN algo. I know that anon account can zap many times but that doesn't count in pushing the zapped comment on top.

**Additional context**
If you want to implement that system of "cowboys credits" you should think very seriously about anon account rights and algo. Anon account must have limited more rights and "influence". Make it hard to be used.

**Comment by ekzyis** [comment]
> As far I understand the "anon" account, if you zap it with x amount of sats, all those sats are going to SN rewards pool. Correct me if I am wrong.

Correct except 5% go to founders as territory revenue but I guess for the sake of the argument, we can ignore that.

> So is kind of meaningless to zap a large amount of sats to "anon" account. Is like rewarding yourself.

It's not meaningless like zaps in general are not meaningless. You zap anon for the same reason you would zap anyone else: to make the post or comment rank higher and gain rewards if you zapped what turned out to be good content at the end of the day.

The only difference with anon is that the sats that would have gone to a stacker go to rewards and thus to the best stackers (which doesn't have to be you and if it is, it has nothing to do with anon). So I don't understand why you have a problem with zapping anon in particular. Fwiw, you can see zapping anon as donations.

Do you have a problem with donations, too?

> If you really want to "approve" that comment, is enough 1 sat zap. I do not understand why people are zapping large amount of sats for something that achieve little.

1 sat = 1 vote, 10 sats = 2 votes, 100 sats = 3 votes etc. (+ votes are weighted by your trust).[^1] So no, it's not enough to zap 1 sat if you think it deserves more (especially compared to other content).

> anon account should pay a fix price to post

Anon pays 100x more to post and comment. Since territories can have different fees, the cost to post is usually way above 1000 sats. For example, it's 100 sats for [~bitcoin](https://stacker.news/~bitcoin) so it would cost anon 10k sats to post something in there.

Comment fee control is planned (#1186) so anon comments might also get more expensive in the future (but I think it's already expensive enough).

> anon account CANNOT zap others comments/posts. I find that as a flaw in SN algo

It's trivial to spin up accounts to workaround this limit so not allowing anon to zap others only increases friction for honest zaps.

What problem are you actually trying to solve? That people can zap themselves via anon even though this does not influence the ranking? It feels like you jumped to solutions without explaining what problem you have.

[^1]: See code for the `weighted_votes_after_tip` function [here](https://github.com/stackernews/stacker.news/blob/master/prisma/migrations/20230506233136_denorm_upvotes/migration.sql) which [gets called in the `item_act` function](https://github.com/stackernews/stacker.news/blob/78520b787b928756c81700ab5b3ac16170ac9084/prisma/migrations/20231226230356_item_act_negative/migration.sql#L33) which [is called if you zap something](https://github.com/stackernews/stacker.news/blob/78520b787b928756c81700ab5b3ac16170ac9084/api/resolvers/item.js#L862-L882).

**Comment by huumn** [comment]
> anon account CANNOT zap others comments/posts. I find that as a flaw in SN algo. I know that anon account can zap many times but that doesn't count in pushing the zapped comment on top.

Anon's zaps only affect ranking when no one trusted is zapping. A 10 sat zap from someone with very low trust will rank higher than an anon zap of 1m sats because anon's have 0 trust.
